packages/lodestar/src/tasks/tasks/cleanUpTask.ts

```diff
@@ -0,0 +1,49 @@
+import {IBeaconChain} from "../../chain";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {ILogger} from "@chainsafe/lodestar-utils";
+import {Epoch} from "@chainsafe/lodestar-types";
+import {IBeaconDb} from "../../db";
+
+export interface ICleanUpModules {
+  chain: IBeaconChain;
+  db: IBeaconDb;
+  logger: ILogger;
+}
+
+/**
+ * Cleanup task to run per epoch or per slot.
+ */
+export class CleanUpTask {
+  private readonly config: IBeaconConfig;
+  private readonly chain: IBeaconChain;
+  private readonly db: IBeaconDb;
+  private readonly logger: ILogger;
+
+  public constructor(config: IBeaconConfig, modules: ICleanUpModules) {
+    this.config = config;
+    this.chain = modules.chain;
+    this.db = modules.db;
+    this.logger = modules.logger;
+  }
+
+  public async start(): Promise<void> {
+    this.chain.clock.onNewEpoch(this.run);
+  }
+
+  public async stop(): Promise<void> {
+    this.chain.clock.unsubscribeFromNewEpoch(this.run);
+  }
+
+  private run = async (epoch: Epoch): Promise<void> => {
+    this.logger.info("Run CleanUpTask at epoch ", epoch);
+    this.logger.profile("CleanUpTask");
+    const indexes = await this.db.activeValidatorCache.values();
+    const headState = await this.chain.getHeadState();
+    await Promise.all(
+      indexes
+        .filter((index) => headState.validators[index].exitEpoch <= epoch)
+        .map((index) => this.db.activeValidatorCache.delete(index))
+    );
```

Comment:
Aren't you suppose to delete all active validators, else validator might disconnect and we still might track him?

Reply:
so I'll have to track the last active epoch of each validator, not just validator index
+ at each epoch, check if lastActiveEpoch < epoch - 1 then delete it

Reply:
We just care about active indexes in current epoch I guess? Not sure what's the best way to do it

Reply:
I can't think of a better way. `EPOCHS_PER_RANDOM_SUBNET_SUBSCRIPTION` = 256 so a subscription last from 256 to 512 so it's not a big difference to know active indexes in previous epoch vs current epoch.